monitoring/monitoring_on.go

```diff
@@ -0,0 +1,67 @@
+// +build monitoring
+
+package monitoring
+
+import (
+	"fmt"
+	"net/http"
+
+	"google.golang.org/grpc"
+	"gopkg.in/macaroon-bakery.v2/bakery"
+
+	"github.com/grpc-ecosystem/go-grpc-prometheus"
+	"github.com/lightningnetwork/lnd/macaroons"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// Enabled signifies whether the monitoring tag was specified when building lnd
+// and whether to automatically export Prometheus metrics.
+const Enabled = true
+
+// GetServerOpts returns the set of interceptors for macaroon authentication
+// (if enabled) and Prometheus monitoring.
+func GetServerOpts(macService *macaroons.Service, perms map[string][]bakery.Op,
+	serverOpts []grpc.ServerOption) []grpc.ServerOption {
+
+	var (
+		unaryInterceptor  grpc.UnaryServerInterceptor
+		streamInterceptor grpc.StreamServerInterceptor
+	)
+
+	// If macaroons aren't disabled (a non-nil service), then we'll set up
```

Comment:
I don't think we need to force the monitoring package to be responsible for setting up the macaroon interceptors. Instead, can't we just through through solely the `serverOpts` then chain them as normal? Applies to the alternate build tag as well.

Reply:
if we gather the unary and stream interceptors separately, we can join them all at once in the rpcserver via: https://github.com/grpc-ecosystem/go-grpc-middleware/blob/master/chain.go#L175

Reply:
IIUC fixed :grimacing: I now gather the unary/stream interceptors separately, and then join them in the rpcserver.